bioimageio/core/cli.py

```diff
@@ -0,0 +1,425 @@
+
+
+class ValidateFormatCmd(CmdBase, WithSource):
+    """bioimageio-validate-format - validate the meta data format of a bioimageio resource."""
+
+    def run(self):
+        validate_format(self.descr)
+
+
+class TestCmd(CmdBase, WithSource):
+    """bioimageio-test - Test a bioimageio resource (beyond meta data formatting)"""
+
+    weight_format: WeightFormatArg = "all"
+    """The weight format to limit testing to.
+
+    (only relevant for model resources)"""
+
+    devices: Optional[Union[str, Sequence[str]]] = None
+    """Device(s) to use for testing"""
+
+    decimal: int = 4
+    """Precision for numerical comparisons"""
+
+    def run(self):
+        test(
+            self.descr,
+            weight_format=self.weight_format,
+            devices=self.devices,
+            decimal=self.decimal,
+        )
+
+
+class PackageCmd(CmdBase, WithSource):
+    """bioimageio-package - save a resource's metadata with its associated files."""
+
+    path: CliPositionalArg[Path]
+    """The path to write the (zipped) package to.
+    If it does not have a `.zip` suffix
+    this command will save the package as an unzipped folder instead."""
+
+    weight_format: WeightFormatArg = "all"
+    """The weight format to include in the package (for model descriptions only)."""
+
+    def run(self):
+        if isinstance(self.descr, InvalidDescr):
+            self.descr.validation_summary.display()
+            raise ValueError("resource description is invalid")
+
+        package(
+            self.descr,
+            self.path,
+            weight_format=self.weight_format,
+        )
+
+
+def _get_stat(
+    model_descr: AnyModelDescr,
+    dataset: Iterable[Sample],
+    dataset_length: int,
+    stats_path: Path,
+) -> Mapping[DatasetMeasure, MeasureValue]:
+    req_dataset_meas, _ = get_required_dataset_measures(model_descr)
+    if not req_dataset_meas:
+        return {}
+
+    req_dataset_meas, _ = get_required_dataset_measures(model_descr)
+    stat_adapter = TypeAdapter(
+        Mapping[DatasetMeasure, MeasureValue],
+        config=ConfigDict(arbitrary_types_allowed=True),
+    )
+
+    if stats_path.exists():
+        logger.info(f"loading precomputed dataset measures from {stats_path}")
+        stat = stat_adapter.validate_json(stats_path.read_bytes())
+        for m in req_dataset_meas:
+            if m not in stat:
+                raise ValueError(f"Missing {m} in {stats_path}")
+
+        return stat
+
+    stats_calc = StatsCalculator(req_dataset_meas)
+
+    for sample in tqdm(
+        dataset, total=dataset_length, descr="precomputing dataset stats", unit="sample"
+    ):
+        stats_calc.update(sample)
+
+    stat = stats_calc.finalize()
+    _ = stats_path.write_bytes(stat_adapter.dump_json(stat))
+
+    return stat
+
+
+class PredictCmd(CmdBase, WithSource):
+    """bioimageio-predict - Run inference on your data with a bioimage.io model."""
+
+    inputs: Union[str, Sequence[str]] = "model_inputs/*/{tensor_id}.*"
```

Comment:
How do you pass a list of inputs?

I have attempted to use
`bioimageio predict affable-shark --inputs input1.npy input2.npy`, but the `input2.npy` isn't recognized. We need argparse for that no?

Reply:
good question, we should probably document that...
https://docs.pydantic.dev/latest/concepts/pydantic_settings/#lists

- `--inputs='[input1.npy,input2.npy']`
- `--inputs input1.npy --inputs input2.npy`
- `--inputs input1.npy,input2.npy`